cli/internal/specs/v0/source.go

```diff
@@ -95,36 +77,15 @@ func (s *Source) UnmarshalSpec(out any) error {
 }
 
 func (s *Source) Validate() error {
-	if s.Name == "" {
-		return fmt.Errorf("name is required")
-	}
-	if s.Path == "" {
-		msg := "path is required"
-		// give a small hint to help users transition from the old config format that didn't require path
-		officialPlugins := []string{"aws", "azure", "gcp", "digitalocean", "github", "heroku", "k8s", "okta", "terraform", "cloudflare"}
-		if funk.ContainsString(officialPlugins, s.Name) {
-			msg += fmt.Sprintf(". Hint: try setting path to cloudquery/%s in your config", s.Name)
-		}
-		return fmt.Errorf(msg)
-	}
-
 	if len(s.Tables) == 0 {
 		return fmt.Errorf("tables configuration is required. Hint: set the tables you want to sync by adding `tables: [...]` or use `cloudquery tables` to list available tables")
 	}
 
-	if s.Registry.NeedVersion() {
-		if s.Version == "" {
-			return fmt.Errorf("version is required")
-		}
-		if !strings.HasPrefix(s.Version, "v") {
-			return fmt.Errorf("version must start with v")
-		}
-	}
 	if len(s.Destinations) == 0 {
 		return fmt.Errorf("at least one destination is required")
 	}
 
-	return nil
+	return s.Metadata.Validate()
```

Comment:
This slightly changes the ordering of the validation steps, let's rather keep it as similar as possible to how it was before and validate `s.Metadata` first, returning an error if it fails

Reply:
76e952ca9debfff063b2632abf4c45e112292bda